### PR TITLE
Adding select language component

### PR DESCRIPTION
**Pull request by SarahBA**

### CODE REVIEW

File: webui/src/components/selectlanguage.jsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        return ret;
+    },
+
+    componentWillMount() {
+        this.componentWillReceiveProps(this.props);
+    },
+
+    componentWillReceiveProps(props) {
+        var def = this.props.defValue || 'eng'; 
+        this.setState({value:def});
```
Comment (SarahBA) [inline webui/src/components/selectlanguage.jsx]: @emanueldima There is a problem here that I couldn't fix. When the user choose a new value, the dropdown menu doesn't show it, while the parent is getting the new value. Can you please take a look?

File: webui/src/components/editrecord.jsx [hunk 2]
```diff
@@ -373,6 +379,22 @@ const EditRecord = React.createClass({
...
+        if (fieldID == "language_code") {
+             return (
+                <div className="form-group row" key={fieldID} style={{marginTop:'1em'}} title={fieldSchema.get('description')}>
+                    <label htmlFor={fieldID} className="col-sm-3 control-label" style={{fontWeight:'bold'}}>
+                        <span style={{float:'right', color:isError?'red':'black'}}>
+                            {fieldSchema.get('title') || fieldID} {type.required ? "*":""}
+                        </span>
+                    </label>
+                    <div className="col-sm-9" >
+                        <SelectLanguage onSelectLanguage={this.handleLanguageCode} defValue={getValue()} />
```
Comment (emanueldima) [inline webui/src/components/editrecord.jsx]: Try using `onSelectLanguage={setValue}`. setValue is defined above and sets the language code directly in the record.

File: webui/src/components/selectlanguage.jsx (hunk 1 above)
Comment (emanueldima) [inline webui/src/components/selectlanguage.jsx]: It's probably related to the setvalue comment, see above.

File: webui/src/components/editrecord.jsx [hunk 3]
```diff
@@ -183,6 +184,7 @@ const EditRecord = React.createClass({
         return {
             record: null,
             fileState: 'done',
+            language: '',
```
Comment (emanueldima) [inline webui/src/components/editrecord.jsx]: This is not needed. The language is set directly in the state.record.

File: webui/src/components/editrecord.jsx [hunk 4]
```diff
@@ -340,6 +342,10 @@ const EditRecord = React.createClass({
         );
     },
 
+    handleLanguageCode: function(langValue) {
```
Comment (emanueldima) [inline webui/src/components/editrecord.jsx]: Not needed, see above.

File: webui/src/data/server.js [hunk 5]
```diff
@@ -522,10 +522,13 @@ class ServerCache {
...
                 },
-                errorFn: (xhr) => this.store.setIn(['languages'], new Error(xhr)),
+                errorFn: (xhr) => {
+                    console.log("Couldn't get the languages: " + xhr)
+                    this.store.setIn(['languages'], new Error(xhr));
+                },                
             });
         }
-        return langs;
+        return this.store.getIn(['languages']);
```
Comment (emanueldima) [inline webui/src/data/server.js]: why the change? it's not needed, and not optimal.

File: webui/src/data/server.js [hunk 6]
```diff
@@ -522,10 +522,13 @@ class ServerCache {
                     console.log('got languages: ', langs.length);
                     this.store.setIn(['languages'], langs);
                 },
-                errorFn: (xhr) => this.store.setIn(['languages'], new Error(xhr)),
+                errorFn: (xhr) => {
+                    console.log("Couldn't get the languages: " + xhr)
```
Comment (emanueldima) [inline webui/src/data/server.js]: try not committing any console.log calls 😄

File: webui/src/components/selectlanguage.jsx (hunk 1 above)
Comment (SarahBA) [inline webui/src/components/selectlanguage.jsx]: The issue was due to using `componentWillReceiveProps`. It doesn't trigger a render after the change!
